jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/bucketActions/CopyAction.kt

```diff
@@ -0,0 +1,13 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+package software.aws.toolkits.jetbrains.services.s3.bucketActions
+
+import com.intellij.icons.AllIcons
+import com.intellij.openapi.project.DumbAware
+import software.aws.toolkits.jetbrains.core.explorer.SingleResourceNodeAction
+import software.aws.toolkits.jetbrains.core.explorer.nodes.AwsExplorerResourceNode
+
+abstract class CopyAction<in T : AwsExplorerResourceNode<*>>(text: String) : SingleResourceNodeAction<T>(text, icon = AllIcons.Actions.Copy),
```

Comment:
I like the idea of an abstract copy action but we should put the logic for interacting with `CopyPasteManager` - the concrete implementation just needs to provide how to get a string from `T`